### ISSUE TITLE
Consider switching to modernc.org/sqlite

### ISSUE THREAD

**Issue by vs49688** [navidrome/navidrome#1934]
### Is your feature request related to a problem? Please describe.

The CGO requirement (from `github.com/mattn/go-sqlite3`) can get rather painful. It'd be good to lessen it, if possible.

### Describe the solution you'd like

Consider switching to `modernc.org/sqlite` -- it's SQLite transpiled to Go, effectively a drop-in replacement.
`sql.Open("sqlite3", ...)` becomes `sql.Open("sqlite", ...)`

### Additional context

* `modernc.org/sqlite` can be slower than `github.com/mattn/go-sqlite3`, but I can't see Navidrome having much database traffic, so this might not be a problem.
* I haven't tested this with Navidrome, but haven't seen any issues with other projects I've migrated.

**Comment by deluan** [comment]
I'd love to use this driver (pure Go!!!) but the performance problem is not when browsing your music, is when importing it, specially the first scan. I need to do some benchmarks to see if this driver does not slow down the scan/import by much.

**Comment by deluan** [comment]
By the way, CGO would still be required for TagLib, and that's the real pain (see the [ci-goreleaser](https://github.com/navidrome/ci-goreleaser/blob/master/Dockerfile) project)

**Comment by vs49688** [comment]
> I'd love to use this driver (pure Go!!!) but the performance problem is not when browsing your music, is when importing it, specially the first scan. I need to do some benchmarks to see if this driver does not slow down the scan/import by much.

Yeah, that was my primary concern with using it. Even if it is a bit slower, there's probably ways to work around it.

> By the way, CGO would still be required for TagLib, and that's the real pain (see the [ci-goreleaser](https://github.com/navidrome/ci-goreleaser/blob/master/Dockerfile) project)

Baby steps... baby steps...
